openwisp_controller/config/base/config.py

```diff
@@ -219,7 +219,12 @@ def manage_vpn_clients(cls, action, instance, pk_set, **kwargs):
         # coming from signal
         if isinstance(pk_set, set):
             template_model = cls.get_template_model()
-            templates = template_model.objects.filter(pk__in=list(pk_set))
+            # Ordering the queryset here doesn't affect the functionality
+            # since pk_set is a set. Though ordering the queryset is required
+            # for tests.
+            templates = template_model.objects.filter(pk__in=list(pk_set)).order_by(
+                'created'
+            )
```

Comment:
are you sure the ordering is required?

Reply:
The bug will only occur when a template for which a VpnClient has already been created comes before the template for which VpnClient does not exist. 

Without ordering the queryset here, the test I wrote was flaky because `set` does not care about order. Ordering the queryset here is just for ensuring that the test works correctly, it is not a part of the fix.